design-toolkit/components/src/components/pagination/index.tsx

```diff
@@ -0,0 +1,218 @@
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+'use client';
+
+import ChevronLeft from '@accelint/icons/chevron-left';
+import ChevronRight from '@accelint/icons/chevron-right';
+import 'client-only';
+import { range } from 'radashi';
+import { createContext, useContext } from 'react';
+import { Button, ToggleButton } from '../button';
+import { Icon } from '../icon';
+import { PaginationStyles } from './styles';
+import type { ProviderProps } from 'react';
+import type {
+  BasePaginationProps,
+  PaginationNavProps,
+  PageNumberContainerProps as PaginationNumberContainerProps,
+  PaginationPageNumberProps,
+  PaginationRange,
+} from './types';
+
+const { container, button } = PaginationStyles();
+const DEFAULT_MIN_RANGE = 1;
+const DEFAULT_MAX_RANGE = 5;
+
+// Return min max range for visible pages. As per our design, we limit
+// the range of numbers to a spread of 5 maximum, getting the lower and upper bounds.
+export function getPaginationRange(
+  pageCount: number,
+  currentPage: number,
+): PaginationRange {
+  if (
+    !(pageCount && currentPage) ||
+    currentPage > pageCount ||
+    pageCount < 1 ||
+    currentPage < 1
+  ) {
+    return { minRange: DEFAULT_MIN_RANGE, maxRange: DEFAULT_MAX_RANGE };
+  }
+
+  // Below max display.
+  if (pageCount < 5) {
+    return {
+      minRange: 1,
+      maxRange: pageCount,
+    };
+  }
+
+  // Middle.
+  if (currentPage >= 3 && currentPage < pageCount - 2) {
+    return {
+      minRange: currentPage - 2,
+      maxRange: currentPage + 2,
+    };
+  }
+
+  // End of page count.
+  if (currentPage > pageCount - 3) {
+    return {
+      minRange: pageCount - 4,
+      maxRange: pageCount,
+    };
+  }
+
+  return { minRange: DEFAULT_MIN_RANGE, maxRange: DEFAULT_MAX_RANGE };
+}
+
+const PaginationContext = createContext<BasePaginationProps>({});
+
+/**
+ *
+ * Pagination - A lightweight implementation for page navigation.
+ *
+ * @example
+ * <Pagination currentPage={1} pageCount={5} onChange={handleOnChange} />
+ */
+export function Pagination({
+  children,
+  classNames,
+  currentPage = 1,
+  pageCount,
+  onChange,
+  ...rest
+}: BasePaginationProps) {
+  return (
+    <Pagination.Provider value={{ currentPage, pageCount }}>
+      <nav
+        {...rest}
+        className={container({ className: classNames?.container })}
+        aria-label='pagination'
+      >
+        <Pagination.Previous
+          onPress={() => onChange?.(currentPage - 1)}
+          className={classNames?.controls}
+        />
+        <Pagination.NumberContainer
+          onPress={(nextPage) => onChange?.(nextPage)}
+          className={classNames?.pages}
+        />
+        <Pagination.Next
+          onPress={() => onChange?.(currentPage + 1)}
```

Comment:
should validate the boundary before mathing

```suggestion
onPress={() => {
  if (currentPage > 1) onChange?.(currentPage + 1);
}}
```